src/compiler.ts

```diff
@@ -255,7 +253,11 @@ export const makeCompiler = (
       if (fs.existsSync(compiledPath)) {
         return
       }
+
       const starTime = new Date().getTime()
+
+      const fileName = params.compile
+      const code = params.code || fs.readFileSync(fileName, 'utf-8')
```

Comment:
We use the passed `params.code`, if available, to avoid reading the file twice. We also do it much later to avoid reading it if it isn’t necessary.